sdk/couchbase/src/durability_level.rs

```diff
@@ -1,49 +1,62 @@
-use serde::Serialize;
+use std::fmt::Display;
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
-#[serde(rename_all = "camelCase")]
-#[non_exhaustive]
-pub enum DurabilityLevel {
-    Majority,
-    MajorityAndPersistToActive,
-    PersistToMajority,
-}
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub struct DurabilityLevel(InnerDurabilityLevel);
 
-impl From<DurabilityLevel> for couchbase_core::memdx::durability_level::DurabilityLevel {
-    fn from(value: DurabilityLevel) -> Self {
-        match value {
-            DurabilityLevel::Majority => {
-                couchbase_core::memdx::durability_level::DurabilityLevel::Majority
-            }
-            DurabilityLevel::MajorityAndPersistToActive => {
-                couchbase_core::memdx::durability_level::DurabilityLevel::MajorityAndPersistToActive
-            }
-            DurabilityLevel::PersistToMajority => {
-                couchbase_core::memdx::durability_level::DurabilityLevel::PersistToMajority
-            }
-        }
+impl DurabilityLevel {
+    pub const NONE: DurabilityLevel = DurabilityLevel(InnerDurabilityLevel::None);
+
+    pub const MAJORITY: DurabilityLevel = DurabilityLevel(InnerDurabilityLevel::Majority);
+
+    pub const MAJORITY_AND_PERSIST_ACTIVE: DurabilityLevel =
+        DurabilityLevel(InnerDurabilityLevel::MajorityAndPersistActive);
+
+    pub const PERSIST_TO_MAJORITY: DurabilityLevel =
+        DurabilityLevel(InnerDurabilityLevel::PersistToMajority);
+
+    pub(crate) fn other(val: String) -> DurabilityLevel {
```

Comment:
Perhaps I'm just missing it, but I don't see where this is used. What is it for?

Reply:
This covers the case where the mgmt API returns a durability level that we don't know about. The choice is this or to return an error. It's not used "right now" but will be shortly when i add the bucket manager to the public API.

Reply:
Makes sense, thanks!